canvas_manage_course/views.py

```diff
@@ -66,29 +66,28 @@ def lti_launch(request):
 def dashboard_course(request):
     course_instance_id = request.LTI.get('lis_course_offering_sourcedid')
 
-    # list of permissions to determine tool visibility
-    tools = {
-        'class_roster': {
-            'visible': lti_permission_required_check(request, settings.CUSTOM_LTI_PERMISSIONS['class_roster'])},
-        'isites_migration': {
-            'visible': lti_permission_required_check(request, settings.CUSTOM_LTI_PERMISSIONS['isites_migration'])},
-        'manage_people': {
-            'visible': lti_permission_required_check(request, settings.CUSTOM_LTI_PERMISSIONS['manage_people'])},
-        'manage_sections': {
-                        'visible': lti_permission_required_check(request, settings.CUSTOM_LTI_PERMISSIONS['manage_sections'])},
-    }
-
-    # django template tags can't do dict lookups, so create a *_visible context
-    # variable for each tool
-    tool_visibility = {
-        '{}_visible'.format(tool): tools[tool]['visible']
-        for tool in tools.keys()}
-
-    view_context = tool_visibility
-
-    # are all tools hidden?
-    no_tools_visible = len(filter(lambda x: x, tool_visibility.values())) == 0
-    view_context['no_tools_visible'] = no_tools_visible
+    tool_access_permission_names = [
+        'class_roster',
+        'im_import_files',  # isites_migration
+        'manage_people',
+        'manage_sections']
+
+    # Verify current user permissions to see the apps on the dashboard
+    allowed = {tool: lti_permission_required_check(request, tool)
+               for tool in tool_access_permission_names}
+    no_tools_allowed = not any(allowed.values())
+
+    view_context = {
+        'allowed': allowed,
+        'no_tools_allowed': no_tools_allowed}
+
+    if no_tools_allowed:
+        view_context['custom_error_title'] = u'Not available'
+        view_context['custom_error_message'] = \
```

Comment:
Maybe use a wrapper template instead of _custom_error.html, to avoid all this boilerplate, as mentioned in the [django-icommons-ui PR](https://github.com/Harvard-University-iCommons/django-icommons-ui/pull/82#discussion_r76648002)